lib/server.js

```diff
@@ -170,12 +170,18 @@ function handleRequest(opts, req, resp) {
     req.headers = req.headers || {};
     req.headers['x-request-id'] = req.headers['x-request-id'] || rbUtil.generateRequestId();
 
+    var xff = req.headers['x-forwarded-for'];
+    var remoteAddr = xff && xff.split(',')[0].trim()
+            || req.socket.remoteAddress;
+    req.headers['x-rb-client-ip'] = remoteAddr;
+
     var reqOpts = {
         conf: opts.conf,
         logger: opts.logger.child({
             req: {
                 method: req.method.toLowerCase(),
-                uri: req.url
+                uri: req.url,
+                headers: req.headers,
```

Comment:
Good point. It's not about stealing but about WM's policy of not keeping sensitive user info around, cf [T112072](https://phabricator.wikimedia.org/T112072)